docs/source/1.0/spec/core/model.rst

```diff
@@ -2692,6 +2692,602 @@ after adding a member to the ``foo`` trait:
+    }
+
+The members are ordered as follows:
+
+1. ``nextToken``
+2. ``pageSize``
+3. ``nameFilter``
+4. ``sizeFilter``
+
+
+Mixins on shapes with non-member properties
+===========================================
+
+Some shapes don't have members, but do have other properties. Adding a mixin
+to such a shape merges the properties of each mixin into the local shape. Only
+certain properties may be defined in the mixin shapes. See the sections below
+for which properties are permitted for each shape type.
+
+Scalar properties defined in the local shape are kept, and non-scalar
+properties are merged. When merging map properties, the values for local keys
+are kept. The ordering of merged lists / sets follows the same ordering as
+members.
+
+Service mixins
+--------------
+
+Service shapes with the :ref:`mixin-trait` may define any property. For
+example, in the following model:
+
+.. code-block:: smithy
+
+    operation OperationA {}
+
+    @mixin
+    service A {
+        version: "A"
+        operations: [OperationA]
+    }
+
+    operation OperationB {}
+
+    @mixin
+    service B with [A] {
+        version: "B"
+        rename: {
+            "smithy.example#OperationA": "OperA"
+            "smithy.example#OperationB": "OperB"
+        }
+        operations: [OperationB]
+    }
+
+    operation OperationC {}
+
+    service C with [B] {
+        version: "C"
+        rename: {
+            "smithy.example#OperationA": "OpA"
+            "smithy.example#OperationC": "OpC"
+        }
+        operations: [OperationC]
+    }
+
+The ``version`` property of the local shape is kept and the ``rename`` and
+``operations`` properties are merged. This is equivalent to the following:
+
+.. code-block:: smithy
+
+    operation OperationA {}
+
+    operation OperationB {}
+
+    operation OperationC {}
+
+    service C {
+        version: "C"
+        rename: {
+            "smithy.example#OperationA": "OpA"
+            "smithy.example#OperationB": "OperB"
+            "smithy.example#OperationC": "OpC"
+        }
+        operations: [OperationA, OperationB, OperationC]
+    }
+
+
+Resource mixins
+---------------
+
+Resource shapes with the :ref:`mixin-trait` MAY NOT define any properties. This
+is because every property of a resource shape is intrinsically tied to its set
+of identifiers. Changing these identifiers would invalidate every other
+property of a given resource. For example:
+
+.. code-block:: smithy
+
+    @mixin
+    @internal
+    resource MixinResource {}
+
+    resource MixedResource with [MixinResource] {}
+
+
+Operation mixins
+----------------
+
+Operation shapes with the :ref:`mixin-trait` MAY NOT define an ``input`` or
+``output`` shape other than the :ref:`unit-type`. This is because allowing
+input and output shapes to be shared goes against the goal of the
+:ref:`input-trait` and :ref:`output-trait`.
+
+Operation shapes with the :ref:`mixin-trait` MAY define the errors shape.
+
+.. code-block:: smithy
```

Comment:
Would be great to provide a slightly better example. Like "ValidatedOperation" or something that defines a ValidationException.